src/olympia/blocklist/tests/test_models.py

```diff
@@ -107,3 +107,49 @@ def test_get_submissions_from_guid(self):
         assert (
             list(BlocklistSubmission.get_submissions_from_guid('ggguid@')) ==
             [])
+
+    def test_all_adu_safe(self):
+        Block.objects.create(
+            addon=addon_factory(guid='zero@adu', average_daily_users=0),
+            updated_by=user_factory())
+        Block.objects.create(
+            addon=addon_factory(guid='normal@adu', average_daily_users=500),
+            updated_by=user_factory())
+        Block.objects.create(
+            addon=addon_factory(guid='high@adu', average_daily_users=999_999),
+            updated_by=user_factory())
+        submission = BlocklistSubmission.objects.create(
+            input_guids='zero@adu\nnormal@adu', min_version=99)
+
+        # not safe because 0 adu
+        submission.to_block = submission._serialize_blocks()
+        assert not submission.all_adu_safe()
+
+        # safe because just normal adu
+        submission.update(input_guids='normal@adu')
+        submission.update(to_block=submission._serialize_blocks())
+        assert submission.all_adu_safe()
+
+        # unsafe because just a high adu addon included
+        submission.update(input_guids='high@adu\nnormal@adu')
+        submission.update(to_block=submission._serialize_blocks())
+        assert not submission.all_adu_safe()
+
+    def test_has_version_changes(self):
+        block = Block.objects.create(
+            addon=addon_factory(guid='guid@'),
+            updated_by=user_factory())
+        submission = BlocklistSubmission.objects.create(
+            input_guids='guid@')
+
+        submission.to_block = submission._serialize_blocks()
+        # no changes to anything
+        assert not submission.has_version_changes()
+
+        block.update(min_version='999', reason='things')
+        # min_version has changed (and reason)
+        assert submission.has_version_changes()
+
+        submission.update(min_version='999')
```

Comment:
Why `block.update()` on line 149 and `submission.update()` here?

Reply:
`block.update` so the submission would be changing the `min_version` back to the default (`0`) and `.has_version_changes()` would be true; `submission.update` then changes it so submission isn't updating the block `min_version` because it's already `'999'` so `.has_version_changes()` would be false; (but it's not the default, so testing it works with non-default properties not changing)